packages/maskbook/src/components/shared/SelectPeopleAndGroups/PersonOrGroupInList.tsx

```diff
@@ -37,62 +37,95 @@ const useStyle = makeStyles((theme: Theme) => ({
 /**
  * Item in the list
  */
-export function ProfileOrGroupInList(props: ProfileOrGroupInListProps) {
-    const { t } = useI18N()
+export const ProfileOrGroupInList = memo<ProfileOrGroupInListProps>((props) => {
     const classes = useStylesExtends(useStyle(), props)
-    const nicknamePreviewsForGroup = useNickNamesFromList(isGroup(props.item) ? props.item.members : [])
-    const listFormat = useIntlListFormat()
 
     const { disabled, ListItemProps: listItemProps, onClick, showAtNetwork } = props
-    let name = ''
-    let avatar: ReturnType<typeof Avatar>
-    let secondaryText: string | undefined = undefined
-    const groupName = useResolveSpecialGroupName(props.item)
-    if (isGroup(props.item)) {
-        const group = props.item
-        name = groupName
-        avatar = (
-            <MuiAvatar>
-                <GroupIcon />
-            </MuiAvatar>
+
+    if (isGroup(props.item))
+        return (
+            <ListItem button disabled={disabled} onClick={onClick} {...listItemProps}>
+                <GroupListItemContent group={props.item} showAtNetwork={showAtNetwork} />
+            </ListItem>
```

Comment:
Let me explain here. Because GroupListItem needs to call `useNickNamesFromList` and `useResolveSpecialGroupName`. But there are time-consuming operations in these two hooks. But the reality is that there are not a lot of Groups. So I extract it to the `GroupListItemContent` component and added a conditional judgment to reduce unnecessary waste. 

![image](https://user-images.githubusercontent.com/28348152/125106013-b7d7d100-e111-11eb-98ef-d6ef0e6834ad.png)